pepsico/comparison.py

```diff
@@ -1,14 +1,82 @@
 import xarray as xr
+import os
+import re
+
+
+ds_zarr = '/data/remic/mydatafiles/zarr/test/hurs'
+ds_nc_dir = '/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global/daily/historical/MPI-ESM1-2-HR/'
+
+
+#only using NetCDF files that contain 'hurs' variable
+def contains_hurs(filename):
+    return 'hurs' in filename
+
+#extracting years from nc file
+def extract_years(filename):
+    match = re.search(r'(\d{4})_(\d{4})\.nc$', filename)
+    if match:
+        start_year = int(match.group(1))
+        end_year = int(match.group(2))
+        return start_year, end_year
+    else:
+        raise ValueError("Unable to find years")
+
+#function that compares nc files(hurs) vs zarred file
+def compare_datasets(nc_file_path, ds_zarr):
+    #returns true if datasets are identical, false otherwise
+
+    #open datasets
+    zarr_dSet = xr.open_zarr(ds_zarr)
+    nc_dSet = xr.open_dataset(nc_file_path)[['hurs']]
+
+    #extracting start and end years from nc file
+    start_year, end_year = extract_years(os.path.basename(nc_file_path))
+
+    time_start = f'{start_year}-01-01'
+    time_end = f'{end_year}-12-31'
+
+    #check if datasets are empty
+    if len(nc_dSet) == 0 or len(zarr_dSet) == 0:
+        print("Datasets are empty")
+        return False
+
+
     
-# defining the path to file  
-ds_zarr = '../data/remic/mydatafiles/zarr/test/'
+    #mapping from zar to nc dimensions (X,Y,T) vs (lon,lat,time)
+    coord_map = {'lon': 'X', 'lat': 'Y', 'time': 'T'}
+
+    #xararay rename function to reset nc labels to zarr
+    nc_dSet = nc_dSet.rename({orig_dim: new_dim for orig_dim, new_dim in coord_map.items() if orig_dim in nc_dSet.dims})
+    nc_dSet = nc_dSet.rename({orig_coord: new_coord for orig_coord, new_coord in coord_map.items() if orig_coord in nc_dSet.coords})
```

Comment:
Would not either one suffice? and then would not the coord case be more general than the dim one?

Reply:
I confirm that either one is enough. In this case all coords are dims and all dims have a coord so the distinction, ast least to rename is meaningless. I would keep coord our of the two.

Reply:
Ok sounds good, I will remove line 49.